dedupe/notebooks/lac_ecr_eda.ipynb

```diff
@@ -0,0 +1,768 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import seaborn as sns\n",
+    "import matplotlib.pyplot as plt\n",
+    "from datetime import datetime\n",
+    "from lxml import etree\n",
+    "from pathlib import Path\n",
+    "from xmldiff import main\n",
+    "from zipfile import ZipFile"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def extract_zip_files(path):\n",
+    "    for file in os.listdir(path):\n",
+    "        if file.endswith(\".zip\"):\n",
+    "            new_dir = os.path.join(path, file[:-4])\n",
+    "            if not os.path.exists(new_dir):\n",
+    "                os.makedirs(new_dir)\n",
+    "                with ZipFile(os.path.join(path, file), 'r') as zip_ref:\n",
+    "                    zip_ref.extractall(new_dir)\n",
+    "\n",
+    "extract_zip_files(\"./../data/LAC_DATA\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "all_dirs = [d for d in os.listdir(\"./../data/LAC_DATA\") if not d.endswith(\".zip\")]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class eCR:\n",
+    "    def __init__(self, path):\n",
+    "        self.path = path\n",
+    "        self.tree = etree.parse(path)\n",
+    "        self.root = self.tree.getroot()\n",
+    "        self.ns = self.root.nsmap\n",
+    "        self.ns['default'] = self.ns.pop(None)\n",
+    "\n",
+    "class RR(eCR):\n",
+    "    def __init__(self, path):\n",
+    "        super().__init__(path)\n",
+    "\n",
+    "class eICR(eCR):\n",
+    "    def __init__(self, path):\n",
+    "        super().__init__(path)\n",
+    "    \n",
+    "    def get_patient_id(self):\n",
+    "        return self.root.find('.//default:patient/default:id', self.ns).attrib['extension']\n",
+    "    \n",
+    "    def get_patient_name(self):\n",
+    "        given = self.root.find('.//default:patient/default:name/default:given', self.ns).text\n",
+    "        family = self.root.find('.//default:patient/default:name/default:family', self.ns).text\n",
+    "        return f\"{given} {family}\"\n",
+    "    \n",
+    "    def get_patient_dob(self):\n",
+    "        return self.root.find('.//default:patient/default:birthTime', self.ns).attrib['value']\n",
+    "    \n",
+    "    def get_send_date(self):\n",
+    "        time = self.root.find('.//default:effectiveTime', self.ns)\n",
```

Comment:
Can we document somewhere our assumptions about data field definitions/mappings? (E.g., what is send date, which XML field is send date, and why)

Reply:
I think this is definitely worth doing!